/*
 * This file was automatically generated by EvoSuite
 * Tue Oct 28 19:43:33 GMT 2025
 */

package com.example;
import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.example.Stack;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class EvoSuiteTest extends EvoSuiteTestScaffolding {

  @Test(timeout = 4000)
  public void test00()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      stack0.push((-1569));
      int int0 = stack0.size();
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test01()  throws Throwable  {
      Stack stack0 = new Stack(914);
      stack0.push(0);
      int int0 = stack0.pop();
      assertEquals(0, stack0.size());
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test02()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      stack0.push(1);
      int int0 = stack0.pop();
      assertEquals(0, stack0.size());
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test03()  throws Throwable  {
      Stack stack0 = new Stack(914);
      stack0.push(0);
      int int0 = stack0.peek();
      assertEquals(1, stack0.size());
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test04()  throws Throwable  {
      Stack stack0 = new Stack(5);
      stack0.push(5);
      int int0 = stack0.peek();
      assertEquals(1, stack0.size());
      assertEquals(5, int0);
  }

  @Test(timeout = 4000)
  public void test05()  throws Throwable  {
      Stack stack0 = null;
      try {
        stack0 = new Stack((-451));
        fail("Expecting exception: NegativeArraySizeException");
      
      } catch(NegativeArraySizeException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("com.example.Stack", e);
      }
  }

  @Test(timeout = 4000)
  public void test06()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      stack0.push((-1569));
      boolean boolean0 = stack0.isEmpty();
      assertEquals(1, stack0.size());
      assertFalse(boolean0);
  }

  @Test(timeout = 4000)
  public void test07()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      stack0.push((-1569));
      int int0 = stack0.peek();
      assertEquals(1, stack0.size());
      assertEquals((-1569), int0);
  }

  @Test(timeout = 4000)
  public void test08()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      stack0.push((-1569));
      int int0 = stack0.pop();
      assertEquals(0, stack0.size());
      assertEquals((-1569), int0);
  }

  @Test(timeout = 4000)
  public void test09()  throws Throwable  {
      Stack stack0 = new Stack(0);
      boolean boolean0 = stack0.isFull();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test10()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      boolean boolean0 = stack0.isFull();
      assertFalse(boolean0);
      assertTrue(stack0.isEmpty());
  }

  @Test(timeout = 4000)
  public void test11()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      boolean boolean0 = stack0.isEmpty();
      assertTrue(boolean0);
  }

  @Test(timeout = 4000)
  public void test12()  throws Throwable  {
      Stack stack0 = new Stack(1691);
      // Undeclared exception!
      try { 
        stack0.peek();
        fail("Expecting exception: IllegalStateException");
      
      } catch(IllegalStateException e) {
         //
         // Stack is empty
         //
         verifyException("com.example.Stack", e);
      }
  }

  @Test(timeout = 4000)
  public void test13()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      // Undeclared exception!
      try { 
        stack0.pop();
        fail("Expecting exception: IllegalStateException");
      
      } catch(IllegalStateException e) {
         //
         // Stack is empty
         //
         verifyException("com.example.Stack", e);
      }
  }

  @Test(timeout = 4000)
  public void test14()  throws Throwable  {
      Stack stack0 = new Stack(0);
      // Undeclared exception!
      try { 
        stack0.push(0);
        fail("Expecting exception: IllegalStateException");
      
      } catch(IllegalStateException e) {
         //
         // Stack is full
         //
         verifyException("com.example.Stack", e);
      }
  }

  @Test(timeout = 4000)
  public void test15()  throws Throwable  {
      String[] stringArray0 = new String[6];
      Stack.main(stringArray0);
      assertEquals(6, stringArray0.length);
  }

  @Test(timeout = 4000)
  public void test16()  throws Throwable  {
      Stack stack0 = new Stack(1407);
      int int0 = stack0.size();
      assertEquals(0, int0);
  }
}
